### ISSUE TITLE
Homogenizing metrics, changing base call.

### ISSUE THREAD

**Issue by cmvcordova** [latent-reasoning-works/manylatents#59]
Two ends

The base Metric definition is currently:

```python
## typing purposes only
from typing import Optional, Protocol

import numpy as np

from src.algorithms.dimensionality_reduction import DimensionalityReductionModule


class Metric(Protocol):
    def __call__(self, 
                 dataset, 
                 embeddings: np.ndarray, 
                 module: Optional[DimensionalityReductionModule] = None
            ) -> float: ...
    
    """A class that defines a metric for evaluating embeddings, irrespective of their modelling source.
    The metric is defined as a callable that takes a dataset and embeddings as input and returns a float value.
    Optionally, a dimensionality reduction module can be passed to the metric, to enable affinity and kernel matrix specific metrics.
    """
``` 
The dataset argument could be passed as optional, as `evaluate_embeddings`, the entry point for the data that then calls these metrics. However, calls in `main` and within specific metrics would have to be updated. Consider adding a third argument, `module`  to be consistent with `evaluate_embeddings`, currently:

```python

@evaluate.register(dict)
def evaluate_embeddings(
    EmbeddingOutputs: dict,
    *,
    cfg: DictConfig,
    datamodule,
    **kwargs,
) -> dict:
    if EmbeddingOutputs is None or EmbeddingOutputs.get("embeddings") is None:
        logger.warning("No embeddings available for evaluation.")
        return {}
    
    if datamodule.mode == "split":
        ds = datamodule.test_dataset
    else:
        ds = datamodule.train_dataset ## defaults to full datset on full runs

    logger.info(f"Reference data shape: {ds.data.shape}")
    logger.info(f"Computing embedding metrics for {ds.data.shape[0]} samples.")
    
    metrics = {}
    all_metrics_cfg = cfg.metrics
    ds_metrics_cfg = all_metrics_cfg.get("dataset", {})
    ds_subsample_fraction = all_metrics_cfg.get("subsample_fraction", None)

    if ds_subsample_fraction is not None:
        ds_subsampled, embeddings_subsampled = subsample_data_and_dataset(ds, EmbeddingOutputs.get("embeddings"), ds_subsample_fraction)
        logger.info(f"Subsampled dataset to {embeddings_subsampled.shape[0]} samples for dataset-level metrics.")
    else:
        ds_subsampled, embeddings_subsampled = ds, EmbeddingOutputs.get("embeddings")

    module = kwargs.get("module", None)
    
    # dataset metrics
    for metric_name, metric_config in ds_metrics_cfg.items():
        metric_fn = hydra.utils.instantiate(metric_config)
        result = metric_fn(ds_subsampled, embeddings_subsampled)
        metrics[metric_name] = result

    # embedding metrics
    embedding_metrics_cfg = all_metrics_cfg.get("embedding", {})
    for metric_name, metric_config in embedding_metrics_cfg.items():
        metric_fn = hydra.utils.instantiate(metric_config)
        # no subsetting here
        result = metric_fn(ds, EmbeddingOutputs.get("embeddings"))
        metrics[metric_name] = result

    # module metrics
    module_metrics_cfg = all_metrics_cfg.get("module", {})
    for metric_name, metric_config in module_metrics_cfg.items():
        metric_fn = hydra.utils.instantiate(metric_config)
        # no subsetting here
        result = metric_fn(ds, EmbeddingOutputs.get("embeddings"), module=module)
        metrics[metric_name] = result

    return metrics
``` 

Additionally, this check - conversion is present across multiple metrics:

```python
    # if using torch tensors:
    if torch.is_tensor(embeddings):
        embeddings = embeddings.cpu().numpy()
        logger.info(f"ParticipationRatio: converted embeddings to numpy, shape {embeddings.shape}")

``` 
Could it be passed directly as a check either in `main` or `evaluate_embeddings `directly, to eliminate redundancies?
